.github/workflows/uploadReleaseArtifacts.yml

```diff
@@ -127,11 +127,9 @@ jobs:
           echo "out=$json" >> $GITHUB_OUTPUT
       - name: Add Build Config for CIRCT-full shared libs
         id: add-build-config-circt-full-shared-libs
+        if: github.event_name == 'release'
```

Comment:
Maybe allow this as workflow_dispatch to support custom invocations? I haven't followed your great CI changes as closely as I'd like but previously limiting things to release sometimes makes it hard to re-run manually.  Whatever you think is best :+1: .

Reply:
I have a feeling it's just the debug info that consumes majorly more space, while the assert space is almost negligible. If possible, I'd like to be able to keep circt-full-shared compiled in Release instead of RelWithDebInfo, so that we have a chance to check the artifacts before releasing them, as well as making it possible for downstreams to test nightly CIRCT libs without having to compile them.

Reply:
@dtzSiFive wrote:
> Maybe allow this as workflow_dispatch to support custom invocations?

The `workflow_dispatch` event for the nightly job is really only for testing. The nightly job is a wrapper that builds a JSON payload that calls the unified build/test/install job. _That latter job_ has a useful `workflow_dispatch` for running whatever you want/need. E.g., I used this this morning to start building 1.51.0 libraries to manually append to the 1.51.0 release (since there's no way to re-run the release without rewriting a tag).

See: https://github.com/llvm/circt/actions/workflows/unifiedBuildTestAndInstall.yml

Options for that are here:
<img width="340" alt="Screenshot 2023-08-18 at 15 36 25" src="https://github.com/llvm/circt/assets/1018530/eda18fd1-8d89-4630-b6a9-9d79e19557fc">

Does that make sense to you?

Reply:
@SpriteOvO 
> If possible, I'd like to be able to keep circt-full-shared compiled in Release instead of RelWithDebInfo, so that we have a chance to check the artifacts before releasing them, as well as making it possible for downstreams to test nightly CIRCT libs without having to compile them.

I can change this to build these if people are going to use them. 👍 I'll use release and asserts off.

Reply:
Thanks @seldridge , that makes sense!  This is great work, wow thank you!

Reply:
I updated the PR to not turn on asserts and relwithdebinfo for any full build.